File: 정민선/프로그래머스/[1차] 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    board = [[' ']*n for _ in range(n)]
    for i in range(n):
        b1num = bin(arr1[i])[::-1][:-2]
        b2num = bin(arr2[i])[::-1][:-2]
        
        for j in range(n):
            if len(b1num) > j and b1num[j] == "1":
                board[i][j] = "#"
                continue
                
            if len(b2num) > j and b2num[j] == "1":
                board[i][j] = "#"
    
    for i in range(n):
        answer.append("".join(board[i][::-1]))
    
    return answer

File: 이진영/프로그래머스/[1차] 비밀지도.js
function solution(n, arr1, arr2) {
    const grid1 = arrToGrid(n, arr1);
    const grid2 = arrToGrid(n, arr2);
    
    let answer = grid1.map((v1, i1) => 
                           v1.map((v2, i2) =>
                                  v2 === 0 && grid2[i1][i2] === 0 ?
                                    " " : "#"
                                 ).join("")
                          )
    
    return answer;
}

const arrToGrid = (n, arr) => {
    return arr.map(num => {
        const bin = num.toString(2).split("").map(Number);
        return [...Array(n - bin.length).fill(0), ...bin];
    });
}